### ISSUE TITLE
Crash when running `:set undofile`

### ISSUE THREAD

**Issue by bps** [rogual/neovim-dot-app#174]
Neovim.app crashes on me when running `:set undofile`. It gets a SIGPIPE when writing `silent! doautoall <nomodeline> FocusLost` to Neovim. I built at the current head of line master (4584aa97151c4c234c8c624e05f1d83ba8936773), and I moved `init.vim` out of the way to help rule out other environmental issues.

Let me know what I can look into if you're unable to reproduce.

```
: bsmyth@ghost ~/src/neovim-dot-app(master); lldb build/Neovim.app/Contents/MacOS/Neovim
(lldb) target create "build/Neovim.app/Contents/MacOS/Neovim"
Current executable set to 'build/Neovim.app/Contents/MacOS/Neovim' (x86_64).
(lldb) run
Process 83734 launched: '/Users/bsmyth/src/neovim-dot-app/build/Neovim.app/Contents/MacOS/Neovim' (x86_64)
```

and I ran `:set undofile` in Neovim.app.

```
Process 83734 stopped
* thread #1: tid = 0x1e07d4b, 0x00007fff99753aaa libsystem_kernel.dylib`write + 10, queue = 'com.apple.main-thread', stop reason = signal SIGPIPE
    frame #0: 0x00007fff99753aaa libsystem_kernel.dylib`write + 10
libsystem_kernel.dylib`write:
->  0x7fff99753aaa <+10>: jae    0x7fff99753ab4            ; <+20>
    0x7fff99753aac <+12>: movq   %rax, %rdi
    0x7fff99753aaf <+15>: jmp    0x7fff9974d414            ; cerror
    0x7fff99753ab4 <+20>: retq   
(lldb) bt
* thread #1: tid = 0x1e07d4b, 0x00007fff99753aaa libsystem_kernel.dylib`write + 10, queue = 'com.apple.main-thread', stop reason = signal SIGPIPE
  * frame #0: 0x00007fff99753aaa libsystem_kernel.dylib`write + 10
    frame #1: 0x00000001000015a8 Neovim`Client::send(this=0x0000000100721c50, message="\x94\0$?vim_command\x91?(silent! doautoall <nomodeline> FocusLost") + 520 at client.cc:25
    frame #2: 0x000000010001953a Neovim`RPC* Client::call<msgpack::v1::type::tuple<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >(this=0x0000000100721c50, name="vim_command", args=tuple<std::__1::basic_string<char> > @ 0x00007fff5fbfc418) + 1450 at client.h:68
    frame #3: 0x0000000100022af6 Neovim`Vim::vim_command(this=0x0000000100721c50, cmd=<unavailable>) + 214 at vim.h:60
    frame #4: 0x000000010004bf9a Neovim`-[VimWindow windowDidResignKey:](self=0x0000000100721920, _cmd="windowDidResignKey:", notification=@"NSWindowDidResignKeyNotification") + 170 at window.mm:181
    frame #5: 0x00007fff96b3fa0c CoreFoundation`__CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
    frame #6: 0x00007fff96b3f99f CoreFoundation`___CFXRegistrationPost_block_invoke + 63
    frame #7: 0x00007fff96b3f917 CoreFoundation`_CFXRegistrationPost + 407
    frame #8: 0x00007fff96b3f682 CoreFoundation`___CFXNotificationPost_block_invoke + 50
    frame #9: 0x00007fff96afc442 CoreFoundation`-[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1922
    frame #10: 0x00007fff96afb695 CoreFoundation`_CFXNotificationPost + 693
    frame #11: 0x00007fff9a6c431a Foundation`-[NSNotificationCenter postNotificationName:object:userInfo:] + 66
    frame #12: 0x00007fff94ff0d5a AppKit`-[NSWindow resignKeyWindow] + 748
    frame #13: 0x00007fff94f186f4 AppKit`_NXSendWindowNotification + 252
    frame #14: 0x00007fff94faba47 AppKit`-[NSWindow _orderOutAndCalcKeyWithCounter:stillVisible:docWindow:] + 377
    frame #15: 0x00007fff94f24713 AppKit`-[NSWindow _reallyDoOrderWindow:relativeTo:findKey:forCounter:force:isModal:] + 2401
    frame #16: 0x00007fff94f22cb2 AppKit`-[NSWindow _doOrderWindow:relativeTo:findKey:forCounter:force:isModal:] + 894
    frame #17: 0x00007fff94f228d6 AppKit`-[NSWindow orderWindow:relativeTo:] + 159
    frame #18: 0x00007fff95096154 AppKit`__18-[NSWindow _close]_block_invoke + 491
    frame #19: 0x00007fff95095f23 AppKit`-[NSWindow _close] + 374
    frame #20: 0x000000010004d7a0 Neovim`-[VimWindow notified:withData:](self=0x0000000100721920, _cmd="notified:withData:", note="neovim.app.nodata", update_o=0x0000700000322c70) + 432 at window.mm:306
    frame #21: 0x0000000100050759 Neovim`-[VimWindow handleEvent:](self=0x0000000100721920, _cmd="handleEvent:", idEvent=0x000000010021af50) + 617 at window.mm:447
    frame #22: 0x00007fff9a73454e Foundation`__NSThreadPerformPerform + 279
    frame #23: 0x00007fff96b49621 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
    frame #24: 0x00007fff96b28e1c CoreFoundation`__CFRunLoopDoSources0 + 556
    frame #25: 0x00007fff96b2833f CoreFoundation`__CFRunLoopRun + 927
    frame #26: 0x00007fff96b27d38 CoreFoundation`CFRunLoopRunSpecific + 296
    frame #27: 0x00007fff91522d55 HIToolbox`RunCurrentEventLoopInMode + 235
    frame #28: 0x00007fff91522b8f HIToolbox`ReceiveNextEventCommon + 432
    frame #29: 0x00007fff915229cf HIToolbox`_BlockUntilNextEventMatchingListInModeWithFilter + 71
    frame #30: 0x00007fff94de0f3a AppKit`_DPSNextEvent + 1067
    frame #31: 0x00007fff94de0369 AppKit`-[NSApplication _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 454
    frame #32: 0x00007fff94dd4ecc AppKit`-[NSApplication run] + 682
    frame #33: 0x0000000100030dfa Neovim`main(argc=1, argv=0x00007fff5fbff9e8) + 186 at main.mm:19
    frame #34: 0x00007fff993775ad libdyld.dylib`start + 1
    frame #35: 0x00007fff993775ad libdyld.dylib`start + 1
(lldb) up
frame #1: 0x00000001000015a8 Neovim`Client::send(this=0x0000000100721c50, message="\x94\0$?vim_command\x91?(silent! doautoall <nomodeline> FocusLost") + 520 at client.cc:25
   22  	/* Send an arbitrary string to the other process's stdin */
   23  	void Client::send(const std::string &message)
   24  	{
-> 25  	    write(process.get_stdin(), &message[0], message.size());
   26  	}
   27  	
   28  	/* Block until something happens */
(lldb) fr v
(Client *) this = 0x0000000100721c50
(const string &) message = "\x94\0$?vim_command\x91?(silent! doautoall <nomodeline> FocusLost": {
  __r_ = {
    std::__1::__libcpp_compressed_pair_imp<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::__rep, std::__1::allocator<char>, 2> = {
      __first_ = {
         = {
          __l = (__cap_ = 65, __size_ = 58, __data_ = "\x94")
          __s = {
             = (__size_ = 'A', __lx = 'A')
            __data_ = {
              [0] = '\0'
              [1] = '\0'
              [2] = '\0'
              [3] = '\0'
              [4] = '\0'
              [5] = '\0'
              [6] = '\0'
              [7] = ':'
              [8] = '\0'
              [9] = '\0'
              [10] = '\0'
              [11] = '\0'
              [12] = '\0'
              [13] = '\0'
              [14] = '\0'
              [15] = '0'
              [16] = '?'
              [17] = ' '
              [18] = '\0'
              [19] = '\x01'
              [20] = '\0'
              [21] = '\0'
              [22] = '\0'
            }
          }
          __r = {
            __words = ([0] = 65, [1] = 58, [2] = 4297116976)
          }
        }
      }
    }
  }
}
(lldb) p process
(Process) $0 = {
  pid = 83794
  pipe_stdin = ([0] = 6, [1] = 8)
  pipe_stdout = ([0] = 9, [1] = 10)
  pipe_stderr = ([0] = 11, [1] = 12)
}
(lldb) ^D
```

**Comment by gleachkr** [comment]
I had a similar issue, which was resolved by neovim/neovim@96dc38b3c84378e5869e030e0a1854e10f2c0a28. It might be worth checking if this helps.

**Comment by bps** [comment]
Indeed it does help. Crash gone. Thanks @gleachkr!
